deploy-service/common/src/main/java/com/pinterest/deployservice/ServiceContext.java

```diff
@@ -102,6 +102,7 @@ public class ServiceContext {
     private Long agentCountCacheTtl;
     private Long maxParallelThreshold;
     private BuildEventPublisher buildEventPublisher;
+    private List<String> accountAllowList;
```

Comment:
This should either be `Collection`, so we accept a wide range of containers,
or `Set` so we can have some performance guarantee.

Reply:
not sure if they are working with yaml configs. https://github.com/spring-projects/spring-boot/issues/1415 Based on the current usages, list is widely used so I'd keep it for now, we can consider move to others if needed.

Reply:
We can convert in the factory. Currently the performance doesn't matter. However, since we are doing this for every ping, and if the allow list grows, then the performance will matter at some point.